projects/plugins/crm/modules/mailpoet/includes/class-mailpoet-background-sync-job.php

```diff
@@ -506,8 +506,12 @@ public function import_subscriber( $subscriber ){
 		//$this->debug( 'Add/Update contact: <pre>' . var_export( $contact_args ) . '</pre>' );
 
 		// Add/update the contact & return id
-		return $zbs->DAL->contacts->addUpdateContact( $contact_args );
-
+		return $zbs->DAL->contacts->addUpdateContact( // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			array(
+				'data'                 => $contact_args['data'],
+				'do_not_update_blanks' => true,
```

Comment:
Two caveats:

* We're now only passing `$contact_args['data']` instead of the entirety of `$contact_args`. At the moment this is okay since there are no other keys in that array, but it could be a bit confusing. An alternative would have been to simply add the `do_not_update_blanks` key around line 417.
* While `do_not_update_blanks` takes care of the issue for now, it would cause problems if MailPoet expanded to have addresses as well. The alternative would be to check for an image in the CRM contact if it exists, and pass that.

Reply:
> While do_not_update_blanks takes care of the issue for now, it would cause problems if MailPoet expanded to have addresses as well. The alternative would be to check for an image in the CRM contact if it exists, and pass that (which I think we do with WooSync or some other integration).

IMHO we have to come up with a better solution (a new function that ties well with our endpoints). This could be done later, we just have to be aware of it.